Upgrade fast_osc to v1.2.0

xavriley opened the pull request:
This commit bumps the version in the vendor folder.

It also adds rake-compiler to the vendored gems to allow compilation to
work for fast_osc. The rake task has to be called from within
`compile-extensions.rb` which looks a little crazy but it's fine,
honest.

It should rescue errors when compiling fast_osc specifically, as it will
fallback to the pure Ruby code in that gem instead so users won't notice
a difference. Other vendored gems with C extensions are unaffected by
this change.

For Windows it also attempts to include `devkit` as a _rake_ dependency
which allows the rake compile task to work, assuming you used Ruby
Installer to install Ruby on your Windows, and that you setup the MSYS2
toolchain (which happens by default).

In short, this shouldn't cause any painful changes and the new features
in fast_osc (potential support for OSC bundles and array args for
supercollider) will work transparently.

samaaron commented:
@xavriley - thanks for this! Just a quick heads up that the Travis build failed.

xavriley commented:
@samaaron thanks! I noticed that - they're passing now.

I've added rake compiler in order to build FastOsc and earlier it was pulling in Rakefiles from wherever the app/server/ruby/bin/compile_extensions.rb script was run from. I've fixed it so that it's only looking in the FastOsc folder now. It's nice to see if green at last!

samaaron commented:
Ace, thanks - now that v3.2 is out we can start playing with this :-)

rbnpi commented:
compile-extensions.rb failed to build this on Raspberry Pi with ruby 2.5.5p157 following Sam's merge of this into master.
